### PR TITLE
Improve checks for diffs

### PR DESCRIPTION
**Pull request by hkdobrev**
Fixes the diff signs on compare pages. See commit messages for details.

### CODE REVIEW

File: extension/content.js [hunk 1]
```diff
@@ -393,7 +393,7 @@ document.addEventListener('DOMContentLoaded', () => {
 				showRealNames();
 			}
 
-			if (pageDetect.isCommit() || pageDetect.isSingleCommit()) {
+			if (pageDetect.hasDiff()) {
```
Comment (hkdobrev) [inline extension/content.js]: This is is something which I wasn't completely sure about. See disclaimer in the commit message and #322. This is not working anyway, but when it is fixed, ideally it should use that check. Feedback is welcome.

File: extension/page-detect.js [hunk 2]
```diff
@@ -29,6 +29,10 @@ window.pageDetect = (() => {
 
 	const isCommit = () => isSingleCommit() || isPRCommit() || (isPRFiles() && $('.full-commit').length > 0);
 
+	const isCompare = () => isRepo() && /^\/compare\//.test(getRepoPath());
+
+	const hasDiff = () => isRepo() && (isSingleCommit() || isPRCommit() || isPRFiles() || isCompare() || (isPR() && $('.diff-table').length > 0));
```
Comment (sindresorhus) [inline extension/page-detect.js]: Yup

**Comment by vvo** [comment]
I have many pages showing "diff supressed" while I want to see it, is this linked? Thanks

**Comment by hkdobrev** [comment]
@vvo Hi! This sounds like a different feature request to expand suppressed diffs automatically, right? What do you mean by "is this linked?"?

**Comment by hkdobrev** [comment]
@sindresorhus Just a quick question before merging this - Do you prefer "Squash and merge" vs. "Rebase and merge"? I've specifically done two separate commits here to separate the fix for a non-working feature with more description in the commit. Similar case here: https://github.com/sindresorhus/refined-github/pull/321 and then the separation is gone https://github.com/sindresorhus/refined-github/commit/23236b2c2517d588a5858a9189fdfd661f70bc4a. Is that how you prefer this? I see you haven't disabled the "Rebase and merge" button in the repo settings and that's why I ask - whether you just haven't disabled it yet or you prefer different merging in different scenarios? Thanks!

**Comment by vvo** [comment]
Sorry I thought refined-github was hiding diff by default. But this is GitHub messing up with me

**Comment by hkdobrev** [comment]
@vvo Yes, GitHub often suppresses large diffs when it would take a lot of time to render the diff with highlighting. Sometimes it suppresses very small diffs on large files. I've noticed that most often with JSON files. You could contact GitHub if you think there's a scenario where a diff shouldn't be suppressed. Or if you think the extension could help with that by expanding diffs automatically in some cases, please open a new issue.

**Comment by sindresorhus** [comment]
`Squash and merge` unless there are independent atomic commits. Rebase is fine in this case.

**Pull request merged**